### DISCUSSION TITLE
uhdr_decompress()

### DISCUSSION THREAD

**Discussion by MajorMurphy** [google/libultrahdr#145] [Q&A] [answered]
This comment mentions calling uhdr_decompress()

https://github.com/google/libultrahdr/blob/ccc54ea3685370f56f182e5c89220028212c9552/ultrahdr_api.h#L561

But I didn't find that function in the API header or sample.
Is the comment referencing a deprecated function?

Thanks

**Comment by gregbenz** [comment, marked as answer]
Must be an outdated name for uhdr_decode().


The decode API options and examples are listed in the app help:

## decode api :
    ultrahdr_app -m 1 -j cosmat_1920x1080_hdr.jpg 
    ultrahdr_app -m 1 -j cosmat_1920x1080_hdr.jpg -o 3 -O 3
    ultrahdr_app -m 1 -j cosmat_1920x1080_hdr.jpg -o 1 -O 5


## decoder options : 
    -j    ultra hdr compressed input resource. 
    -o    output transfer function, optional. [0:linear, 1:hlg (default), 2:pq, 3:srgb] 
    -O    output color format, optional. [3:rgba8888, 4:rgbahalffloat, 5:rgba1010102 (default)] 
          It should be noted that not all combinations of output color format and output 
          transfer function are supported. 
          srgb output color transfer shall be paired with rgba8888 only. 
          hlg, pq shall be paired with rgba1010102. 
          linear shall be paired with rgbahalffloat. 

## common options : 
    -z    output filename, optional. 
          in encoding mode, default output filename 'out.jpeg'. 
          in decoding mode, default output filename 'outrgb.raw'.

**Reply by MajorMurphy** [reply]
That's what I expected, but glad to have it confirmed.  Thanks!

**Reply by gregbenz** [reply]
Just to be clear, I'm speculating. I haven't been writing any of this code myself. But given it's in the JSDOC for that function, seems very likely to be the case.
